### PR TITLE
(feat) : add extended user roles and privileges to support facility wide users

### PR DESCRIPTION
**Pull request by donaldkibet**
## Description

This PR adds extended roles and privileges as specified in this document [Excel Role and Privileges](https://palladiumgroup-my.sharepoint.com/:x:/r/personal/eric_njeru_thepalladiumgroup_com/Documents/Desktop/Role-based%20access%20control(RBAC)%20for%20facility%20wide%20EMR.xlsx?d=w8a0a8deccadc4b2fad366ddf8adca151&csf=1&web=1&e=T98Wgo)

@ErickSomie @mmatheka @RAJABIBRAZ @Murithijoshua Please give this a try and report where the system isn't configured correctly. 

### Setup instructions

1. Ensure you have copied the `configuration` folder from this branch and deleted `configuration_checksum`folder to `\var\lib\OpenMRS` deployment folder.
2. Restart your server and ensure `Initializer module` has installed the new **Roles and Privileges** 
3. Login in with Admin rights and create users as specified in the excel document e.g
     for a receptionist/registration clerk. Create user named receptionist under the roles assign `o3: Receptionist/Registration` role. Do not add additional roles to the user. I have under the hood taken care of API related privileges for all users.
     
 **NB**
Only assign one role to a user. if a user is a **Nurse** only assign that role thanks

### CODE REVIEW

**Review by Murithijoshua — APPROVED** [review]

**Comment by donaldkibet** [comment]
cc @RAJABIBRAZ @ErickSomie are we good with the configuration done

File: configuration/privileges/privileges.csv [hunk 1]
```diff
@@ -1,42 +1,10 @@
...
-6687fb9c-d9b5-4b40-b1ff-464cf8067f29,o3: View MCH,,
-ec9651aa-c4c8-4d0a-8cad-826dd7ca5eb1,o3: View HTS,,
-c359a69a-1efb-4145-804b-d961340de309,o3: View Default Tracing,,
-092f11ae-2029-4f97-a278-29ce499d7fe9,o3: View Family History,,
-7dc14272-97cf-4637-9e1d-3d45ae53f7fc,o3: View Service Queues,,
-3a537438-80c7-4774-b86a-8ca70ac2e572,o3: View Appointments,,
-664dbb64-4737-4201-93aa-b00ddb2943e5,o3: View Billing,,
+50e9f584-8950-44f0-9cb8-72b96fd82f5b,o3: View Home,,
+afb6fda5-6e59-4106-9379-0684e9bc9284,o3: View Active Visits,,
+cdf12bea-1e6e-40a8-bd21-ff287fdb22d9,o3: View Patient Search,,
```
Comment (makombe) [inline configuration/privileges/privileges.csv:4]: @donaldkibet  now that you have grouped privileges into specific apps, could we also do the same here and delete `privileges.csv` file completely?

**Comment by ErickSomie** [comment]
From my end what I tested we need to look at the landing pages for the
cashier and the lab person.

**Comment by Murithijoshua** [comment]
@donaldkibet & @ErickSomie I noticed for Receptionist s/he has ability to view orders/order basket but cannot record despite icon to add a order being available, My question is, for a receptionist, is he allowed to view this, even if it read only?  
Screenshot 2024-08-13 at 11 35 08

**Comment by donaldkibet** [comment]
> @donaldkibet & @ErickSomie I noticed for Receptionist s/he has ability to view orders/order basket but cannot record despite icon to add a order being available, My question is, for a receptionist, is he allowed to view this, even if it read only? Screenshot 2024-08-13 at 11 35 08

This should be disabled for the registration clerk

File: configuration/privileges/privileges.csv (hunk 1 above)
Comment (donaldkibet) [reply configuration/privileges/privileges.csv:4]: Sure will update accordingly

**Comment by makombe** [comment]
@andrineM @RAJABIBRAZ @enyaencha @Murithijoshua I will merge this in for you to be able to test it appropriately. Thanks

**Pull request merged**